packages/components/src/core/Icon/style.ts

```diff
@@ -9,6 +9,7 @@ export interface IconExtraProps<IconName extends keyof IconNameToSizes>
   extends CommonThemeProps {
   className?: string;
   sdsIcon: IconName;
+  svgIcon?: FC<CustomSVGProps>;
```

Comment:
something for after the Hackday! But I wonder if there's a way to express that sdsIcon and svgIcon are mutually exclusive and at least one is required 😄 Maybe something to ask ChatGPT lol

Or perhaps we could combine these two props to `icon` or something better that accepts `IconName | FC<CustomSVGProps>`, and the component can check at runtime, if it's `IconName` or `FC` and act accordingly?

What do you think?

Reply:
Yea I definitely think that makes the dx a lot better than how it's set up right now but ts was complaining a whole bunch 😂

Reply:
Ahaha yeah no doubt! TS is such a huge pain sometimes 😆 

@masoudmanson would love to get your input on this API design too 💡 🙏 

Thanks all!

Reply:
I had to undo the modifications for the Icon component. This component is specifically crafted to render SDS icons, expecting an SDS Icon name and returning a JSX Element. While switching the `sdsIcon` prop to `icon` seemed like a reasonable change, I believe there is no urgent need to make this change and force lots of breaking changes to the product teams!